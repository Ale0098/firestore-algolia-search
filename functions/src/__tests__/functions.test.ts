const { logger } = require("firebase-functions");

const consoleInfoSpy = jest.spyOn(logger, "info").mockImplementation();

import functionsConfig from "../config";
import { obfuscatedConfig } from "../logs";
import * as exportedFunctions from "../";

describe("extension", () => {
  // beforeEach(() => {});

  test("functions configuration detected from environment variables", async () => {
    expect(functionsConfig).toMatchSnapshot();
  });

  test("functions configuration is logged on initialize", async () => {
    expect(consoleInfoSpy).toBeCalledWith(
      "Initializing extension with configuration",
      obfuscatedConfig
    );
  });

  test("functions are exported", async () => {
    expect(exportedFunctions.executeIndexOperation).toBeInstanceOf(Function);
  });
});
