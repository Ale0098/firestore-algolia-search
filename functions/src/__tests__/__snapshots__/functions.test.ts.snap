// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extension functions configuration detected from environment variables 1`] = `
Object {
  "algoliaAPIKey": "",
  "algoliaAppId": "",
  "algoliaIndexName": "",
  "collectionPath": "",
  "fields": "",
}
`;
